x-pack/legacy/plugins/infra/public/utils/datemath.ts

```diff
@@ -0,0 +1,226 @@
+/*
```

Comment:
Where'd this file come from? Looks like a lot of low-level functionality, is this something modified from one of our core libs?

Reply:
Not modified, but built on top of them.

I think these functions should be in `@elastic/datemath` itself, but before doing that I want to make sure I get the functions right.